src/index.test.ts

```diff
@@ -271,6 +289,29 @@ describe('jsonFetch', function () {
 
       throw new Error('Should have failed');
     });
+
+    it('call the onRequestStart and onRequestEnd functions in each retry', async function () {
+      const onRequestStart = sandbox.spy(() => {});
+      const onRequestEnd = sandbox.spy(() => {});
+
+      try {
+        await jsonFetch('foo.bar', {
+          shouldRetry: () => true,
+          retry: {
+            retries: 5,
+            factor: 0,
+          },
+          onRequestStart,
+          onRequestEnd,
+        });
+      } catch {
+        expect(onRequestStart.callCount).to.equal(6);
+        expect(onRequestEnd.callCount).to.equal(6);
```

Comment:
it would be nice to actually have confidence that the first call passes in `retryCount: 0`, then `retryCount: 1`, etc. Maybe assert this? For simplicity, you can lower `retries: 5` to `retries: 2` or something.

Reply:
I am unable to recreate this, could you please suggest an approach?

Reply:
see comment above

```js
expect(onRequestStart).to.have.been.calledWithMatch({retryCount: 1});
expect(onRequestStart).to.have.been.calledWithMatch({retryCount: 2});
expect(onRequestStart).to.have.been.calledWithMatch({retryCount: 3});
```

We can pair if this isn't working.